### ISSUE TITLE
Support as const in no-object-literal-type-assertion

### ISSUE THREAD

**Issue by aboyton** [typescript-eslint/typescript-eslint#166]
Sorry for not using the template, it didn't seem to apply in this case. This ticket is related to #66. Feel free to close if you think it is covered by other tickets.

TypeScript have merged in a new feature `as const` (slated for TypeScript 3.4, see https://github.com/Microsoft/TypeScript/pull/29510). This should get around many of the times where I need to say things like:

```ts
return {
  type: ChunkContents.LINK,
  content: linkMatch[2],
  ref: linkMatch[1],
} as {
  type: ChunkContents.LINK;
  content: string;
  ref: string;
};
```

It would be great to get the rule `no-object-literal-type-assertion` to support `as const` but disallow `as someOtherType` (at least as an option).

**Comment by bradzacher** [comment]
ohh i didn't know they were adding that! That's an awesome feature!
